### PR TITLE
lightning: make default file router support compression

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/38514

Problem Summary:

### What is changed and how it works?
- [x] Lightning file router default regular support for file name parsing compression
- [x] Lightning disables splitting of large files in compressed CSV files
Implement two tasks in compress lightning

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- None

Documentation

- None

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/mydump/region.go [hunk 1]
```diff
@@ -283,7 +284,8 @@ func makeSourceFileRegion(
 	// We increase the check threshold by 1/10 of the `max-region-size` because the source file size dumped by tools
 	// like dumpling might be slight exceed the threshold when it is equal `max-region-size`, so we can
 	// avoid split a lot of small chunks.
-	if isCsvFile && cfg.Mydumper.StrictFormat && dataFileSize > int64(cfg.Mydumper.MaxRegionSize+cfg.Mydumper.MaxRegionSize/largeCSVLowerThresholdRation) {
+	if isCsvFile && cfg.Mydumper.StrictFormat && dataFileSize > int64(cfg.Mydumper.MaxRegionSize+cfg.Mydumper.MaxRegionSize/largeCSVLowerThresholdRation) &&
```
Comment (hawkingrei) [inline br/pkg/lightning/mydump/region.go:287]: This line is too long.

File: br/pkg/lightning/mydump/router.go [hunk 2]
```diff
@@ -128,15 +136,15 @@ var expandVariablePattern = regexp.MustCompile(`\$(?:\$|[\pL\p{Nd}_]+|\{[\pL\p{N
...
-	// db schema create file pattern, matches files like '{schema}-schema-create.sql'
-	{Pattern: `(?i)^(?:[^/]*/)*([^/.]+)-schema-create\.sql$`, Schema: "$1", Table: "", Type: SchemaSchema, Unescape: true},
-	// table schema create file pattern, matches files like '{schema}.{table}-schema.sql'
-	{Pattern: `(?i)^(?:[^/]*/)*([^/.]+)\.(.*?)-schema\.sql$`, Schema: "$1", Table: "$2", Type: TableSchema, Unescape: true},
-	// view schema create file pattern, matches files like '{schema}.{table}-schema-view.sql'
-	{Pattern: `(?i)^(?:[^/]*/)*([^/.]+)\.(.*?)-schema-view\.sql$`, Schema: "$1", Table: "$2", Type: ViewSchema, Unescape: true},
-	// source file pattern, matches files like '{schema}.{table}.0001.{sql|csv}'
-	{Pattern: `(?i)^(?:[^/]*/)*([^/.]+)\.(.*?)(?:\.([0-9]+))?\.(sql|csv|parquet)$`, Schema: "$1", Table: "$2", Type: "$4", Key: "$3", Unescape: true},
+	{Pattern: `(?i).*(-schema-trigger|-schema-post)\.sql(?:\.(\w*?))$`, Type: "ignore"},
+	// db schema create file pattern, matches files like '{schema}-schema-create.sql{.compress}'
```
Comment (lance6716) [inline br/pkg/lightning/mydump/router.go:140]:
```suggestion
	// db schema create file pattern, matches files like '{schema}-schema-create.sql.{compress}'
```

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: br/pkg/lightning/mydump/router.go (hunk 2 above)
Comment (lichunzhu) [reply br/pkg/lightning/mydump/router.go:140]: How about this? I want to express that this compress suffix (including the dot '.') is optional. 
```suggestion
	// db schema create file pattern, matches files like '{schema}-schema-create.sql(.compress)'
```

Comment (lance6716) [reply br/pkg/lightning/mydump/router.go]: I now prefer `{schema}-schema-create.sql[.{compress}]`

Comment (lichunzhu) [reply br/pkg/lightning/mydump/router.go]: addressed

**Review by lance6716 — APPROVED** [review]

**Review by lyzx2001 — APPROVED** [review]
lgtm

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**